src/orchard/java.clj

```diff
@@ -44,58 +42,27 @@
 ;; functions that modify the classpath. You can find a more extensive discussion
 ;; on that topic here https://github.com/clojure-emacs/orchard/issues/103.
 
-(defn jdk-find
+(defn ^:deprecated jdk-find
   "Search common JDK path configurations for a specified file name and return a
   URL if found. This accommodates `java.home` being set to either the JDK root
   (JDK9+) or a JRE directory within this (JDK 8), and searches both the home and
   `lib` directories."
-  [f]
-  (let [home (io/file (System/getProperty "java.home"))
-        parent (.getParentFile home)
-        paths [(io/file home f)
-               (io/file home "lib" f)
-               (io/file parent f)
-               (io/file parent "lib" f)]]
-    (->> paths (filter #(.canRead ^File %)) first io/as-url)))
-
-(def add-java-sources-via-dynapath?
-  "Should orchard use the dynapath library to use \"fetch Java sources\" functionality?
-
-  Note that using dynapath currently implies some bugs, so you might want to disable this option."
-  (contains? #{"true" "1"} (System/getProperty "orchard.use-dynapath" "true")))
-
-(def jdk-sources
-  "The JDK sources path. If found on the existing classpath, this is the
-  corresponding classpath entry. Otherwise, the JDK directory is searched for
-  the file `src.zip`."
-  (when add-java-sources-via-dynapath?
-    (let [base-url (fn [path]
-                     (some-> (io/resource path)
-                             ^JarURLConnection (. openConnection)
-                             (. getJarFileURL)))]
-      (or (base-url "java.base/java/lang/Object.java") ; JDK9+
-          (base-url "java/lang/Object.java")           ; JDK8-
-          (jdk-find "src.zip")))))
+  [_]
+  nil)
+
+(def ^:deprecated jdk-sources nil)
```

Comment:
From what I pondered on previous iterations: ^:deprecated + no-op impls are the least breaking choice

(as opposed to an `(assert false)`)